libraries/stdlib/genosl/stdlib_genosl_impl.mtlx

```diff
@@ -704,18 +704,18 @@
   <implementation name="IM_convert_integer_float_genosl" nodedef="ND_convert_integer_float" target="genosl" sourcecode="float({{in}})" />
 
   <!-- <combine2> -->
-  <implementation name="IM_combine2_vector2_genosl" nodedef="ND_combine2_vector2" target="genosl" />
-  <implementation name="IM_combine2_color4CF_genosl" nodedef="ND_combine2_color4CF" target="genosl" />
-  <implementation name="IM_combine2_vector4VF_genosl" nodedef="ND_combine2_vector4VF" target="genosl" />
-  <implementation name="IM_combine2_vector4VV_genosl" nodedef="ND_combine2_vector4VV" target="genosl" />
+  <implementation name="IM_combine2_vector2_genosl" nodedef="ND_combine2_vector2" target="genosl" sourcecode="{ {{in1}},{{in2}} }"/>
```

Comment:
Do these implementations compile successfully in OSL?  I was expecting to see type-specific constructors such as `vector2` and `color4` in these implementations, though I haven't yet tested this pull request with an OSL build.

Reply:
I believe that OSL allows the use of initializer lists, and the full test suite passes - I was assuming that that would exercise the osl compiler to validate?

certainly happy to revert to type'd constructors if thats preferred.  I do see that might be clearer to read.

Reply:
On closer inspection, it does appear that the CI is not running `oslc` to validate the OSL source code.  I wonder if this could be changed? And due to a mixup on my side, I wasn't running it locally either.  This uncovered both syntax errors here, as well as in the already merged #1854, so I'll put a PR to fix that asap as well.

Reply:
Thanks for these fixes, and we're very open to ideas on improving the integration of OSL into render testing.  On my side, I currently set the flags described here when running OSL render tests, and there's definitely room to simplify this:

https://github.com/AcademySoftwareFoundation/MaterialX/tree/main/source/MaterialXTest#per-language-render-setup

Reply:
Do you think it's possible to get these to run in the CI for the PRs?  I'm not familiar with the MaterialX GitHub CI setup.

Reply:
The main issue is that we only allow source code to be used in GitHub CI if we can build it from scratch, since downloading binaries creates potential security issues.  And building Open Shading Language from scratch takes a bit too long for us to include this in continuous builds.  One could imagine including OSL in a nightly or weekly build of MaterialX, though, if we end up creating such as build type in the future.

Reply:
Another possibility would be to leverage one of the VFX Platform build environments, as described in this issue:
https://github.com/AcademySoftwareFoundation/MaterialX/issues/1382

If the VFX Platform build environment already includes OSL, then we should be able to safely reference this in our render test suite.